src/Mod/Mesh/App/Core/MeshIO.cpp

```diff
@@ -1656,31 +1658,88 @@ bool MeshInput::LoadNastran (std::istream &rstrIn)
 
     while (std::getline(rstrIn, line)) {
         upper(ltrim(line));
-        if (line.find("GRID*") == 0) {
-        }
-        else if (line.find('*') == 0) {
-        }
-        // insert the read-in vertex into a map to preserve the order
-        else if (line.find("GRID") == 0) {
+        if (line.empty()) {
+            // Skip all the following tests
+        }
+        else if (line.rfind("GRID*", 0) == 0) {
+            // This element is the 16-digit-precision GRID element, which occupies two lines of the card. Note that
+            // FreeCAD discards the extra precision, downcasting to an four-byte float.

```

Comment:
Well, mostly because I didn't want to rewrite the whole class 😃 . I left the code sort of split out, though, so if someone actually cared about that feature in the future it would be easy modify the input.

Reply:
I see. Maybe if there is reason to support higher precision in the future.

Reply:
Long format is quite common now.

Reply:
I can definitely imagine that in FEM, a lot of work is done in purely double precision (certainly in CFD we had switched almost entirely to double already over a decade ago, for both grids and solution data). I'm just unsure of the use-case in the Mesh WB -- I think we need to see who actually uses the NASTRAN input here, now that it actually works right!

Reply:
I am quite interested in using it to evaluate the 2D mesh quality and maybe tweak the mesh a bit. And it can export back, That is fantastic.